slang.h

```diff
@@ -3950,11 +3955,23 @@ namespace slang
             LayoutRules     rules = LayoutRules::Default,
             ISlangBlob**    outDiagnostics = nullptr) = 0;
 
-        virtual SLANG_NO_THROW TypeLayoutReflection* SLANG_MCALL getParameterBlockLayout(
+            /** Get a container type from `elementType`. For example, given type `T`, returns
+                a type that represents `StructuredBuffer<T>`.
+
+                @param `elementType`: the element type to wrap around.
+                @param `containerType`: the type of the container to wrap `elementType` in.
+                @param `outDiagnostics`: a blob to receive diagnostic messages.
+            */
+        virtual SLANG_NO_THROW TypeReflection* SLANG_MCALL getContainerType(
```

Comment:
I think of this operation as a short-term necessary evil. Ideally the user should be able to do something like:

```
auto structuredBufferGeneric = session->getStandardLibraryDecl("StructuredBuffer");
auto structuredBufferOfMyType = structuredBufferGeneric->specialize({myType});
```

I get why that's not easy to support right now.

One smaller question would be whether `getContainerType` could use a string name for the container type instead of an enum:

```
auto structuredBufferOfMyType = ...->getContainerType("StructuredBuffer", myType);
```

Technically all of the types in that `enum` are named types in the Slang standard library...